infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -600,60 +493,55 @@ public void processLockRelease() {
         stopLogReplication();
         isLeader.set(false);
         // Signal Log Replication Server/Sink to stop receiving messages, leadership loss
-        if (localClusterDescriptor != null && localClusterDescriptor.getRole() == ClusterRole.SINK) {
+        if (topologyDescriptor.getLocalClusterDescriptor() != null && topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SINK) {
             interClusterServerNode.setLeadership(false);
         }
         recordLockRelease();
     }
 
     /**
-     * Process Topology Config Change:
+     * Process topology config change:
      * - Higher config id
      * - Potential cluster role change
-     * <p>
-     * Cluster change from Source to Sink is a two step process, we first
-     * confirm that
-     * we are ready to do the cluster role change, so by the time we receive cluster change
-     * notification, nothing needs to be done, other than stop.
+     *
+     * Cluster change from Source to Sink is a two step process,
+     * we first confirm that we are ready to do the cluster role change,
```

Comment:
We do process all updates, not just a role change.  Take a look at processTopologyChangeNotification().  If the role has not changed, the default handling is in onSinkClusterAddRemove() which invokes sessionManager.refresh().
It will update the topology config id.

onSinkClusterAddRemove() can be renamed to something more generic in a follow-up PR when we support different topologies.  Will add a TODO for it.

Resolving this comment.  Please reopen it if you do not agree.

Reply:
Got it, so higher config id is not considered as `onClusterRoleChange()` right? it will be handled by `onSinkClusterAddRemove` by default. Maybe we could remove the "Higher config id" in this method's javadoc, I was reading from that...

Reply:
yes, that's right.  This method is specifically for role change.  Updated the comment.